library/chemistry/src/JordanWigner/Convenience.qs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export
+    TrotterStepOracle,
+    QubitizationOracle,
+    OptimizedQubitizationOracle;
+
+import JordanWigner.JordanWignerBlockEncoding.JordanWignerBlockEncodingGeneratorSystem;
+import JordanWigner.JordanWignerEvolutionSet.JordanWignerFermionEvolutionSet;
+import JordanWigner.JordanWignerEvolutionSet.JordanWignerGeneratorSystem;
+import JordanWigner.JordanWignerOptimizedBlockEncoding.JordanWignerOptimizedBlockEncoding;
+import JordanWigner.JordanWignerOptimizedBlockEncoding.PauliBlockEncoding;
+import JordanWigner.JordanWignerOptimizedBlockEncoding.QuantumWalkByQubitization;
+import JordanWigner.StatePreparation.PrepareArbitraryStateD;
+import JordanWigner.Utils.JordanWignerEncodingData;
+import JordanWigner.Utils.MultiplexOperationsFromGenerator;
+import JordanWigner.Utils.TrotterSimulationAlgorithm;
+import Std.Convert.IntAsDouble;
+import Std.Math.Ceiling;
+import Std.Math.Lg;
+import Utils.EvolutionGenerator;
+import Utils.GeneratorSystem;
+
+// Convenience functions for performing simulation.
+
+/// # Summary
+/// Returns Trotter step operation and the parameters necessary to run it.
+///
+/// # Input
+/// ## qSharpData
+/// Hamiltonian described by `JordanWignerEncodingData` format.
+/// ## trotterStepSize
+/// Step size of Trotter integrator.
+/// ## trotterOrder
+/// Order of Trotter integrator.
+///
+/// # Output
+/// A tuple where: `Int` is the number of qubits allocated,
+/// `Double` is `1.0/trotterStepSize`, and the operation
+/// is the Trotter step.
+function TrotterStepOracle(qSharpData : JordanWignerEncodingData, trotterStepSize : Double, trotterOrder : Int) : (Int, (Double, (Qubit[] => Unit is Adj + Ctl))) {
```

Comment:
Here and in other functions. qSharpData identifier doesn't describe this parameter well. Maybe jwData? Also probably better to use individual fields rather than splitting into fields with !. statePrepData and energyShift aren't used here.
This applies to other functions. Also, this is just a small issue, we can discuss if we need to change this. I can push things like this into your branch.

Reply:
I prepared a PR.